src/schema_test_tools.py

```diff
@@ -0,0 +1,106 @@
+        except YAMLError as exc:
+            warnings.warn('Failed to open ' + filename + ' as YAML')
+    return y
+
+
+def get_validator(filename, base_uri=''):
+    """Load schema from JSON file;
+    Check whether it's a valid schema;
+    Return a Draft4Validator object.
+    Optionally specify a base URI for relative path
+    resolution of JSON pointers (This is especially useful
+    for local resolution via base_uri of form file://{some_path}/)
+    """
+
+    schema = get_json_from_file(filename)
+    try:
+        # Check schema via class method call. Works, despite IDE complaining
+        # However, it appears that this doesn't catch every schema issue.
+        Draft4Validator.check_schema(schema)
+        print("%s is a valid JSON schema" % filename)
+    except SchemaError:
+        raise
+    if base_uri:
+        resolver = RefResolver(base_uri=base_uri,
+                               referrer=filename)
+    else:
+        resolver = None
+    return Draft4Validator(schema=schema,
+                           resolver=resolver)
+
+
+def validate(validator, instance):
+    """Validate an instance of a schema and report errors."""
+    if validator.is_valid(instance):
+        print("Validation Passes")
+        return True
+    else:
+        es = validator.iter_errors(instance)
+        recurse_through_errors(es)
+        print("Validation Fails")
+        return False
+
+
+def recurse_through_errors(es, level=0):
+    """Recurse through errors posting message
+    and schema path until context is empty"""
+    # Assuming blank context is a sufficient escape clause here.
+    for e in es:
+        warnings.warn(
+            "***" * level + " subschema level " + str(level) + "\t".join([str(e.message),
+                                                                          "Path to error:" + str(
+                                                                              e.absolute_schema_path)]) + "\n")
+        if e.context:
+            level += 1
+            recurse_through_errors(e.context, level=level)
+
+
+def test_local(path_to_schema_dir, schema_file, test_dir, load_yaml=True):
+    """Tests all instances in a test_folder against a single schema.
+    Assumes all schema files in single dir.
+    Assumes all *.json files in the test_dir should validate against the schema.
+       * path_to_schema_dir:  Absolute or relative path to schema dir
+       * schema_file: schema file name
+       * test_dir: path to test directory (absolute or local to schema dir)
+    """
+    file_ext = 'json'
+    loader = get_json_from_file
+    if load_yaml:
+        file_ext = 'yaml'
+        loader = get_yaml_from_file
+#    os.chdir(path_to_schema_dir)
+    pwd = subprocess.check_output('pwd').decode("utf-8").rstrip()
+    # Removed resolver as this was causing failure of local path resolution.
+    # base_uri = "file://" + pwd + "/" + path_to_schema_dir
+    sv = get_validator(path_to_schema_dir + '/' + schema_file) #, base_uri)
```

Comment:
With following test code;
```
test_local(path_to_schema_dir='config/prod/',
           schema_file='neo4j2owl_config_schema.json',
           test_dir='config/prod/') 
```
`sv` become `config/prod//neo4j2owl_config_schema.json`, should we change the code or the test code? Or we can just add a check whether the directory ends with '/' or not and act accordingly.

Reply:
Same for following line with `glob.glob(pathname=test_dir + '/*.' + file_ext)`

Reply:
Happy for you to fix.